src/buffer.rs

```diff
@@ -252,6 +270,24 @@ impl Buffer {
             self.data.insert(start + i, c)
         }
     }
+
+    /// Check if the other buffer starts with the same content as this one.
+    /// Used to implement autosuggestions.
+    pub fn is_match(&self, other: &Buffer) -> bool {
```

Comment:
Can you think of a more descriptive name? From a glance, it's a bit difficult to tell what this does. Thanks